### PR TITLE
Update CODEOWNERS

### PR DESCRIPTION
**Pull request by AdamTumi**
New scheme for CODEOWNERS file.

### CODE REVIEW

**Review by jane-intel — COMMENTED** [review]
Hello! 
Your branch is hanging in the main repository of OV -- where we only store release branches. Could you please merge it and delete the branch? Proper way of doing it is to have fork and create branches there and then pull changes from your fork and branch to OV
image

**Comment by ilyachur** [comment]
@AdamTumi As I see at the current moment CODEOWNERS file contains some errors:
Screenshot 2022-08-24 at 14 12 08
Can we solve these errors before the merge?

File: CODEOWNERS
Comment (AdamTumi) [reply CODEOWNERS]: Yes, thanks for spotting this. Next time I'll follow the rules for PR creation.

File: CODEOWNERS
Comment (ilyachur) [inline CODEOWNERS:88]: I think we still need to have `/docs/template_plugin/` and also `/docs/template_extension/`

File: CODEOWNERS
Comment (ilyachur) [inline CODEOWNERS:49]: Why did you remove these paths?

File: CODEOWNERS
Comment (ilyachur) [inline CODEOWNERS:58]: The same for vpu

File: CODEOWNERS
Comment (AdamTumi) [reply CODEOWNERS:88]: Good catch. I'll fix it.

File: CODEOWNERS
Comment (AdamTumi) [reply CODEOWNERS:49]: These paths are not listed in the reference PPT - that's why I removed them.

File: CODEOWNERS
Comment (AdamTumi) [reply CODEOWNERS:58]: The same as for removed gpu paths.

File: CODEOWNERS
Comment (ilyachur) [reply CODEOWNERS:49]: But we have these paths at the current moment. I don't want to change this logic. It is a legacy gpu part

File: CODEOWNERS
Comment (ilyachur) [reply CODEOWNERS:58]: The same comment as for gpu

Comment (AdamTumi) [reply CODEOWNERS:58]: Reverted this change.

File: CODEOWNERS
Comment (AdamTumi) [reply CODEOWNERS]: Reverted this change.

File: CODEOWNERS
Comment (ilyachur) [inline CODEOWNERS:66]: I think we also need to add vpu tests

File: CODEOWNERS
Comment (ilyachur) [inline CODEOWNERS:75]: I think we also need to revert this change

**Review by ilyachur — APPROVED** [review]
In general LGTM some minor comments.
And one more common comment.
At the current moment majority of maintainer groups have `-ie-` part, I think we can remove `ie` from group names.

**Pull request merged**
